### ISSUE TITLE
Large storyboards cause stack overflow on iOS Safari

### ISSUE THREAD

**Issue by blueberrymuffin3** [blueberrymuffin3/osu-js#11]
Issue loading https://osu-js.pages.dev/play/?1006822: `Error loading beatmap: RangeError: Maximum call stack size exceeded`

![screenshot](https://user-images.githubusercontent.com/43014304/180740387-afa79616-1c53-405b-9d2c-10f973ef5b86.png)

**Comment by kionell** [comment]
Let me tell you a cool story about how i tried to find the reason of this problem. I set up a free trial iOS emulator only to be able to get this error. Since i didn't have the opportunity to debug this, I simply had to display the error stack in the progress bar.
![image](https://user-images.githubusercontent.com/43014304/182019016-322ad3b3-a52a-4233-8890-77c522f533c6.png)

This led me to the line 2069 with the following push method:

![image](https://user-images.githubusercontent.com/43014304/182019185-3a9d0498-3299-4f99-b476-d4a875b10b1f.png)

Which corresponds to the next line in the code:

https://github.com/blueberrymuffin3/osu-storyboard-parser/blob/main/src/storyboard.ts#L29
